ha/setup/ha_setup.py

```diff
@@ -570,21 +572,42 @@ def _update_node_map(self) -> None:
         """
         Update node map
         """
-        machine_id = self.get_machine_id()
-        node_id = Conf.get(self._index, f"server_node.{machine_id}.node_id")
-        cluster_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.CLUSTER_ID.value}")
-        site_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.SITE_ID.value}")
-        rack_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.RACK_ID.value}")
-        storageset_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.STORAGESET_ID.value}")
-        node_map = {NODE_MAP_ATTRIBUTES.CLUSTER_ID.value: cluster_id, NODE_MAP_ATTRIBUTES.SITE_ID.value: site_id,
-                    NODE_MAP_ATTRIBUTES.RACK_ID.value: rack_id, NODE_MAP_ATTRIBUTES.STORAGESET_ID.value: storageset_id}
-        system_health = SystemHealth(self._confstore)
-        key = system_health._prepare_key(const.COMPONENTS.NODE_MAP.value, node_id=node_id)
-        # Check key is already exist if not, store the node map.
-        node_map_val = self._confstore.get(key)
-        if node_map_val is None:
-            self._confstore.set(key, str(node_map))
-
+        try:
+            machine_id = self.get_machine_id()
+            node_id = Conf.get(self._index, f"server_node.{machine_id}.node_id")
+            cluster_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.CLUSTER_ID.value}")
+            site_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.SITE_ID.value}")
+            rack_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.RACK_ID.value}")
+            storageset_id = Conf.get(self._index, f"server_node.{machine_id}.{NODE_MAP_ATTRIBUTES.STORAGESET_ID.value}")
+            node_map = {NODE_MAP_ATTRIBUTES.CLUSTER_ID.value: cluster_id, NODE_MAP_ATTRIBUTES.SITE_ID.value: site_id,
+                        NODE_MAP_ATTRIBUTES.RACK_ID.value: rack_id, NODE_MAP_ATTRIBUTES.STORAGESET_ID.value: storageset_id}
+            system_health = SystemHealth(self._confstore)
+            key = system_health._prepare_key(const.COMPONENTS.NODE_MAP.value, node_id=node_id)
+            # Check key is already exist if not, store the node map.
+            node_map_val = self._confstore.get(key)
+            if node_map_val is None:
+                self._confstore.set(key, str(node_map))
+
+            # TBD: Temporary code till EOS-17892 is implemented
+            initial_health = EntityHealth()
```

Comment:
Please put this in different function, a private one.

Reply:
@ajaysrivas  It is a temporary code which will be deleted soon. I needed some info e.g. cluster_id, site_id, etc. which is available already in this method, hence added it here. I can move, but is it really needed?

Reply:
yes @indrajitzagade. If this code is separate, disabling/removing the code will be better. There will be less probability in making any error.

Reply:
Done @ajaysrivas